backend/src/daemon/fwupdmgr.rs

```diff
@@ -0,0 +1,128 @@
+use std::process::Command;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+#[allow(non_snake_case)]
+struct Devices {
+    pub Devices: Vec<Device>,
+}
+
+impl Devices {
+    pub fn get_launch(self) -> Result<Device, Error> {
+        self.Devices
+            .into_iter()
+            .find(|device| {
+                device.name().unwrap_or(&"".to_string()).contains("Launch")
+                    && device.vendor().unwrap_or(&"".to_string()) == "System76"
+            })
+            .ok_or(Error::NoLaunch)
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[allow(non_snake_case)]
+struct Device {
+    Name: Option<String>,
+    Vendor: Option<String>,
+    Created: usize,
+    Releases: Option<Vec<Release>>,
+}
+
+impl Device {
+    pub fn name(&self) -> Option<&String> {
+        self.Name.as_ref()
+    }
+
+    pub fn vendor(&self) -> Option<&String> {
+        self.Vendor.as_ref()
+    }
+
+    pub fn created(&self) -> usize {
+        self.Created
+    }
+
+    pub fn is_newer_release(self) -> Result<bool, Error> {
+        Ok(self
+            .Releases
+            .as_ref()
+            .ok_or(Error::NoRelease)?
+            .into_iter()
+            .any(|release| release.created() > self.created()))
```

Comment:
Hm, does device `Created` attribute really have the right meaning for this to work? running `fwupdmgr get-devices --json`, it looks like the Unix time corresponds to today (on my not recently updated launch), I guess when `fwupdmgr` fetched the information from the device.

The version of the firmware can be retrieved through `system76_ectool`.  I don't think the build date is exposed anywhere by the firmware, unless something changed in the newer QMK version.

Reply:
I'm mostly just using it for sorting, so the day doesn't have to be exect, but I'm not sure there is a better attribute for this?

Reply:
I would also rather not use `system76_ectool` for this because production has asked that the keyboard configurator eventually updates firmware itself. I think it makes the most sense to pull the available versions from the same tool used to update the firmware.

Reply:
`system76_ectool` is used as a library to communicate with the keyboard already, so it might avoid a dependency if there's an easy way to fetch the latest version from LVFS, but yeah, I guess that makes sense if this is later going to be used to update the firmware.

Comparing the version `Version` might make more sense, if it's just necessary to test if the version is exactly equal. Or more simply, fwupd already knows how to compare versions, and it looks like the `is-upgrade` field is set based on that?